src/shogun/evaluation/CrossValidation.cpp

```diff
@@ -264,32 +264,57 @@ float64_t CCrossValidation::evaluate_one_run()
 		m_machine->set_store_model_features(true);
 
 		/* do actual cross-validation */
+		#pragma omp parallel for if (get_global_parallel()->get_num_threads()!=1)
 		for (index_t i=0; i <num_subsets; ++i)
 		{
+			CMachine* machine;
+			CFeatures* features;
+			CLabels* labels;
+
+			if (get_global_parallel()->get_num_threads()==1)
+			{
+				machine=m_machine;
+			}
+			else
+			{
+				machine=(CMachine*)m_machine->clone();
```

Comment:
Wondering why this worked.
https://github.com/shogun-toolbox/shogun/blob/develop/src/shogun/base/SGObject.cpp#L721
clone REFs the copy. So no REF needed ?